python/taichi/profiler/kernelprofiler.py

```diff
@@ -0,0 +1,194 @@
+        #clear backend & frontend
+        impl.get_runtime().prog.clear_kernel_profile_info()
+        self.clear_fronted()
+
+    def update_records(self):
+        """Acquires profiling records from a backend."""
+        impl.get_runtime().sync()
+        self.clear_fronted()
+        self._traced_records = impl.get_runtime(
+        ).prog.get_kernel_profiler_records()
+
+    def count_results(self):
+        """Counts the statistical results.
+
+        Profiling records with the same kernel name will be counted in a instance of class:`~taichi.profiler.kernelprofiler.StatisticalResult`.
+        Presenting kernel profiling results in a statistical perspactive.
+        """
+        for record in self._traced_records:
+            if self._statistical_results.get(record.name) == None:
+                self._statistical_results[record.name] = StatisticalResult(
+                    record.name)
+            self._statistical_results[record.name].insert_record(
+                record.kernel_time)
+            self._total_time_ms += record.kernel_time
+        self._statistical_results = {
+            k: v
+            for k, v in sorted(self._statistical_results.items(),
+                               key=lambda item: item[1],
+                               reverse=True)
+        }
+
+    # print info mode
+    COUNT = 'count'  # print the statistical results (min,max,avg time) of Taichi kernels.
+    TRACE = 'trace'  # print the records of launched Taichi kernels with specific profiling metrics (time, memory load/store and core utilization etc.)
+
+    def print_info(self, mode=COUNT):
+        """Print the profiling results of Taichi kernels.
+
+        To enable this profiler, set ``kernel_profiler=True`` in ``ti.init()``.
+        The default print mode is ``'count'`` mode: print the statistical results (min,max,avg time) of Taichi kernels,
+        another mode ``'trace'``: print the records of launched Taichi kernels with specific profiling metrics (time, memory load/store and core utilization etc.)
+
+        Args:
+            mode (str): the way to print profiling results
+        """
+        def patition_line(character, num):
+            return character * num
+
+        self.update_records()  # trace records
+        self.count_results()  # statistical results
+
+        #count mode (default) : print statistical results of all kernel
+        if mode == self.COUNT:
+            print(f"{patition_line('=',73)}"\
+                  f"\n{_ti_core.arch_name(ti.cfg.arch).upper()} Profiler(count)\n"\
+                  f"{patition_line('=',73)}")
+            print(f"[      %     total   count |"\
+                  f"      min       avg       max   ]"\
+                  f" Kernel name\n")
+            for key in self._statistical_results:
+                result = self._statistical_results[key]
+                fraction = result.total_time / self._total_time_ms * 100.0
+                print(f"["
```

Comment:
nit: use """ to wrap them in one block so that it's easier to read as a whole ;) 
>>> message = f"""
...     Hi {name}. 
...     You are a {profession}. 
...     You were in {affiliation}.
... """
...
>>> message
'\n    Hi Eric.\n    You are a comedian.\n    You were in Monty Python.\n'

Reply:
Sorry, I need to display these results in one line, multiple lines introduce spaces and indents that are not easily eliminated.

Reply:
But inspired by this, part of the code was refactored in 9af12f7
```
            table_header = f"""
                {patition_line('=',73)}
                {_ti_core.arch_name(ti.cfg.arch).upper()} Profiler(count)
                {patition_line('=',73)}
            """
            items_header = f"""
                [      %     total   count |      min       avg       max   ] Kernel name
            """
            print(inspect.cleandoc(table_header))
            print(inspect.cleandoc(items_header))
```

Reply:
Sorry, it was my code that caused the misunderstanding, I have changed this part back to `“...{}...”.format(...)`  e9a84ea